calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java

```diff
@@ -103,12 +103,16 @@ public String importStreamFromDelimitedFile(
 				Component vTimeZone = calendar.getComponent("VTIMEZONE");
 				if (vTimeZone!=null) {
 					Property tzProperty = vTimeZone.getProperty("TZID");
-					if (tzProperty!=null) calendarTzid = tzProperty.getValue();
+					if (tzProperty!=null) {
+						calendarTzid = tzProperty.getValue();
+						ZoneId.of(calendarTzid);
```

Comment:
Perhaps a comment to tell people this is just to validate the time zone, that it throws the exception.

Reply:
Makes sense, done!